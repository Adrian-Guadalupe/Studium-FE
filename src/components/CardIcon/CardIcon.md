```js
<CardIcon  cardNumber="5"/>
```