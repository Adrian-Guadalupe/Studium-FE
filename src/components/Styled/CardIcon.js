import React, { useState } from 'react';
import styled from 'styled-components';
import { Card } from 'antd';

const CardIcon = ({cardNumber}) => {
    return (
        <div style = {{position: "absolute"}}>
    <StyledStackedCardIconTwo></StyledStackedCardIconTwo>
        <StyledStackedCardIcon></StyledStackedCardIcon>
            <StyledCardIcon style = {{width:"40px", height:"47px"}}>
                <CardIconText>{cardNumber.number}</CardIconText>
            </StyledCardIcon>
            
        </div>
        )
    }

const StyledCardIcon = styled (Card)`
&& {
    display:flex;
    width: 40px;
    height: 52px;
    left: 316px;
    top: 110px;
    background: #C4C4C4;
    border: 1px solid #FFFFFF;
    box-sizing: border-box;
    }
`

const StyledStackedCardIcon = styled (Card)`
&& {
    position: absolute;
    width: 42px;
    height: 53px;
    left: 321px;
    top: 102px;
    background: #C4C4C4;
    border: 1px solid #FFFFFF;
    box-sizing: border-box;
}
`
const StyledStackedCardIconTwo = styled (Card)`
&& {
position: absolute;
width: 42px;
height: 56px;
left: 328px;
top: 94px;
background: #C4C4C4;
border: 1px solid #FFFFFF;
box-sizing: border-box;
}
`

const CardIconText = styled.div`
position: absolute;
top: 12px;
left: 36%;
font-family: Fredoka One;
font-style: normal;
font-weight: 900;
font-size: 30px;
line-height: 18px;
text-align: center;
color: #000000; 
`
export default CardIcon;