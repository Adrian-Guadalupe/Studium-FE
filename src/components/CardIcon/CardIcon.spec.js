import React from 'react';
import renderer from 'react-test-renderer';

import {CardIcon} from './CardIcon.js';

test( 'CardIcon render correctly', () => {
  const tree = renderer.create( <CardIcon/> ).toJSON();
  expect( tree ).toMatchSnapshot();
} );