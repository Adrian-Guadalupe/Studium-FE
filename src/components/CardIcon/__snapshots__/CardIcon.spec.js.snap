// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardIcon render correctly 1`] = `
.c0 {
  margin-top: 4rem;
  position: relative;
  width: 80px;
  height: 108px;
}

.c3 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 80px;
  height: 108px;
  left: 0;
  bottom: 0;
  background: #c4c4c4;
  border: 3px solid white;
}

.c2 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 80px;
  height: 108px;
  left: 0;
  bottom: 0;
  background: #c4c4c4;
  border: 3px solid white;
  left: 20%;
  bottom: 15%;
}

.c1 {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 80px;
  height: 108px;
  left: 0;
  bottom: 0;
  background: #c4c4c4;
  border: 3px solid white;
  left: 40%;
  bottom: 30%;
}

.c4 {
  position: absolute;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  font-family: Fredoka One;
  font-style: normal;
  font-weight: 900;
  font-size: 30px;
  line-height: 18px;
  text-align: center;
  color: #000000;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2"
  />
  <div
    className="c3"
  />
  <div
    className="c4"
  />
</div>
`;
